src/MapfileStyleParser.ts

```diff
@@ -314,24 +320,100 @@ export class MapfileStyleParser implements StyleParser {
       markSymbolizer.strokeWidth = parseFloat(styleParameters.width);
     }
 
-    // TODO: Convert hack for demo to proper SYMBOL handling!
-    const wellKnownName = styleParameters.symbol;
-    switch (wellKnownName) {
-    case 'circle':
+    // TODO: Abstract to SVG and drop WellKnownName!
+    const symbolType = styleParameters.symbol.type.toLowerCase();
+    switch (symbolType) {
+    case 'ellipse': {
+      const xy = styleParameters.symbol.points[0].split(' ');
+      if (xy[0] === xy[1]){
+        markSymbolizer.wellKnownName = 'Circle' as WellKnownName;
+      }
+      break;
+    }
     case 'square':
     case 'triangle':
     case 'star':
     case 'cross':
     case 'x': {
-      const wkn = wellKnownName.charAt(0).toUpperCase() + wellKnownName.slice(1);
-      markSymbolizer.wellKnownName = wkn as WellKnownName;
       break;
     }
     }
 
     return markSymbolizer;
   }
 
+  /**
+   * Get the GeoStyler-Style IconSymbolizer from an Mapfile Style
+   *
+   * @param {object} styleParameters The Mapfile Style
+   * @return {IconSymbolizer} The GeoStyler-Style IconSymbolizer
+   */
+  getIconSymbolizerFromMapfileStyle(styleParameters: any): IconSymbolizer {
```

Comment:
Can you merge some lines in a common function for getIconSymbolizerFromMapfileStyle and getTextSymbolizerFromMapfileStyle ?

Reply:
Done in #10